🐛 BUG: Command failed: git rev-parse --abbrev-ref HEAD

### What version of `Wrangler` are you using?

2.0.22

### What operating system are you using?

Windows 10

### Describe the Bug

- Create new empty directory
- Initialize Git repository in that directory with `git --init`
- Run `wrangler pages project create xyz` inside the new directory

Result:

```
X [ERROR] Command failed: git rev-parse --abbrev-ref HEAD

  fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
  Use '--' to separate paths from revisions, like this:
  'git <command> [<revision>...] -- [<file>...]'
```

--- a/.changeset/orange-ads-marry.md
+++ b/.changeset/orange-ads-marry.md
@@ -1,0 +1,5 @@
+---
+"wrangler": patch
+---
+
+fix(#1487) [pages]: Command failed: git rev-parse --abrev-ref HEAD

--- a/packages/wrangler/src/pages/projects.tsx
+++ b/packages/wrangler/src/pages/projects.tsx
@@ -124,12 +124,18 @@ export async function CreateHandler({
 			isGitDir = false;
 		}
 
+		if (isGitDir) {
+			try {
+				productionBranch = execSync(`git rev-parse --abbrev-ref HEAD`)
+					.toString()
+					.trim();
+			} catch (err) {}
+		}
+
 		productionBranch = await prompt(
 			"Enter the production branch name:",
 			"text",
-			isGitDir
-				? execSync(`git rev-parse --abbrev-ref HEAD`).toString().trim()
-				: "production"
+			productionBranch ?? "production"
 		);
 	}
 

